portal-kernel/src/com/liferay/portal/kernel/util/PortalUtil.java

```diff
@@ -1750,11 +1750,11 @@ public static PortletURL getSiteAdministrationURL(
 	}
 
 	public static String getSiteAdminURL(
-			Company company, Group group, String ppid,
+			ThemeDisplay themeDisplay, Group group, String ppid,
 			Map<String, String[]> params)
 		throws PortalException {
-
-		return getPortal().getSiteAdminURL(company, group, ppid, params);
+		Portal portal = getPortal();
+		return themeDisplay.getPortalURL() + portal.getSiteAdminURL(null, group, ppid, params);
```

Comment:
I don't think we should implement the logic at here

Reply:
Also, can you make sure it does not have side effects? Is it possible to introduce a new method?